### DISCUSSION TITLE
Can I simulate/test remote access when VLC and the Slingbox Server are on the same LAN as the Slingbox?

### DISCUSSION THREAD

**Discussion by DVoorhees1957** [GerryDazoo/Slinger#219] [Q&A]
Right now all of my (admittedly limited) testing has been with the VLC app, the slingbox server and the Slingbox all being located on the same local LAN. I'd like to do some testing of my intended configuration, with the VLC app and the slingbox server code running on my laptop but accessing the Slingbox via the external IP of my router. What port or ports do I need to open in my router - just the Slingbox port (5206 in my current case), or do I need to open and forward other ports as well? And do I need to do something special to make my laptop look like it's not on the same LAN as the Slingbox?

BTW, I don't need to send any commands to the Remote; I'm using a different solution for that, which sends commands directly to my TiVo(s).

**Comment by DVoorhees1957** [comment]
Well, that was a bust... what am I doing wrong? I want to simulate a "remote access" attempt, even though my laptop (running VLC and the slingbox_server) is still on the same LAN as the Slingbox. I modified the ipaddress entry in the config.ini file to be that of my router. Slingbox_server starts up ok, but I can't figure out what URL to give VLC to complete the "remote access" test; when I use my local address in VLC (i.e., http://192.168.1.124:8080/slingbox/sb500), I get a timeout error. The URL is supposed to be targeting my slingbox_server (my laptop) not my Slingbox, right?

**Reply by RandomMan719** [reply]
Yes, VLC points to the server ip address and server port number, in your case the laptop.  The Slingbox IP address and ports are in the config.ini. As well as the server port.
I think the issue is you can't just point the server to your router and say it's the Slingbox.

**Reply by DVoorhees1957** [reply]
So there's no way for me to put my laptop "outside of the local LAN" (as far as my router is concerned)? While that makes some logical sense, it also sorta... well, disappoints me.

**Reply by RandomMan719** [reply]
I have multiple remote sites with a slingbox at each site.  At each site I have a router, in the remote routers I port forward an external port to the ip address and internal port of the Slingbox. The external port can be whatever you want.  Like 6000 and it points to port 5206 and the ip address of slingbox.  So in my config.ini for that remote slingbox I put in the public ip address of the remote site(should be a static public ip address or else it will change on you) and the port 6000 (or whatever external port you set) for my slingbox. Name the slingbox in the config as well, let's say I name it SiteA

In the config.ini you also set the server port which can be whatever you want as well, let's say 7000 is the server port. You can also set the URL base in the config, let's say the URLbase=Gerry
In VLC I would put in this URL http://ipaddressofserver(your  laptop):7000(Server port)/Gerry(URLbase)/SiteA(Slingbox name)

Things in the parentheses just tell you what each part of the url points to, don't use any parentheses in the URL.

Hopefully that helps.

**Reply by RandomMan719** [reply]
You could probably port forward an external port to your sling box on your lan.  Find out your public ip address, just Google what's my IP address.  Then set your config.INI to point to your public ip address and external port for that slingbox.  Then take your laptop and hook it up to someone else's Internet or if you can do a mobile hotspot. That would make it like your local lan is a remote site and the server would be on the outside of your lan.

And you connect to your Slingbox through the internet.

**Reply by RandomMan719** [reply]
Also if you are using VLC on the same laptop as the server set vlc to point to http://127.0.0.1 for the ipaddress, this points to the laptop's ip address even if it changes, because if you connect the laptop to another network outside of yours, that network will give the laptop/server a different ip address.

**Reply by RandomMan719** [reply]
![image](https://user-images.githubusercontent.com/116921195/199632055-7d890a5a-0224-412c-82de-249de72939b6.jpeg)

**Reply by RandomMan719** [reply]
This is an example, I have the server running on this laptop, and my server port is 8080, my URLbase=Gerry, Slingbox name is BoysenVerizon.

**Reply by DVoorhees1957** [reply]
Thanks for the 127.0.0.1 tip! I probably wouldn't have thought of that. I'm going to have 2 basic configs, the current one (mostly for testing) that has VLC and Slingbox_Server both running on the laptop, and the eventual one (which I can't test right now) which will have MX Player or VLC running on a Fire TV Stick, Slingbox Server running on my laptop - both located on the same LAN - and my Slingbox and TiVo combo running on a completely different LAN.

Also, I realized I don't have to work to hard to do an actual remote test - just drive to the neighborhood Starbucks and fire up Slingbox_Server and VLC on my laptop...

**Comment by gbjbasdw** [comment]
I just spotted your post and was interested in your comment "BTW, I don't need to send any commands to the Remote; I'm using a different solution for that, which sends commands directly to my TiVo(s)."

What are you doing to achieve this ?

**Reply by DVoorhees1957** [reply]
See discussion #398. Basically, I'm using a Broadlink RM4, which is a universal/learning remote that is also networked, so I can communicate with the RM4 via an iPhone app from anywhere in the world, and the RM4 (which sits in my living room pointing at my 2 TiVos) sends whatever commands I want directly to the TiVos, without involving either Slinger or my Slingboxes...
